src/pages/admin/milestone.tsx

```diff
@@ -4,27 +4,37 @@ import {
   VerticalFormItem,
   VerticalFormTitle,
 } from '@/components/common';
-import { config } from '@/lib';
-import withAccessType from '@/lib/hoc/withAccessType';
+import { showToast, config } from '@/lib';
+import { AdminAPI } from '@/lib/api';
+import withAccessType, { GetServerSidePropsWithAuth } from '@/lib/hoc/withAccessType';
 import { PermissionService } from '@/lib/services';
-import type { GetServerSideProps, NextPage } from 'next';
+import type { NextPage } from 'next';
 import { SubmitHandler, useForm } from 'react-hook-form';
-import { AiOutlineMail } from 'react-icons/ai';
+import { AiOutlineStar } from 'react-icons/ai';
 import { VscLock } from 'react-icons/vsc';
 
 interface FormValues {
   name: string;
   points: number;
 }
-const AwardPointsPage: NextPage = () => {
+
+interface AwardPointsPageProps {
+  authToken: string;
+}
+const AwardPointsPage: NextPage<AwardPointsPageProps> = ({ authToken }) => {
   const {
     register,
     handleSubmit,
     formState: { errors },
   } = useForm<FormValues>();
 
-  const onSubmit: SubmitHandler<FormValues> = () => {
-    // TODO
+  const onSubmit: SubmitHandler<FormValues> = async ({ name, points }) => {
+    try {
+      await AdminAPI.createMilestone(authToken, name, Number(points));
+      showToast('Successfully awarded attendance!');
+    } catch (error) {
+      showToast('Error found!');
```

Comment:
You should use `reportError` from utils to display the error from the API:

```suggestion
      reportError('Error found!', error);
```

Reply:
that way here it would say

> **Error found!**
>
> invalid input syntax for type integer: "0.5"

which is a bit helpful than just this:

![image](https://github.com/user-attachments/assets/385f62d2-b08c-4ca0-98ab-fe5c9a604b1e)